core/message_pool.go

```diff
@@ -7,25 +7,50 @@ import (
 	"github.com/ipfs/go-cid"
 	"github.com/pkg/errors"
 
+	"github.com/filecoin-project/go-filecoin/actor"
 	"github.com/filecoin-project/go-filecoin/address"
 	"github.com/filecoin-project/go-filecoin/chain"
+	"github.com/filecoin-project/go-filecoin/consensus"
 	"github.com/filecoin-project/go-filecoin/metrics"
+	"github.com/filecoin-project/go-filecoin/state"
```

Comment:
Blocking: these dependencies are really unfortunate. The message pool is a low level component, nearly a leaf of an ideal dependency tree. This is bringing too much into scope.

Instead, inject the validation logic from outside. The consensus package already, necessarily has this set of deps and other validation logic. Test the validation logic independently of the pool, and for the pool merely test that the pool respects the decision of a (fake) validator.